buildpack/bp_descriptor.go

```diff
@@ -6,13 +6,38 @@ import (
 	"path/filepath"
 
 	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
 )
 
 type BpDescriptor struct {
-	WithAPI     string `toml:"api"`
-	Buildpack   BpInfo `toml:"buildpack"`
-	Order       Order  `toml:"order"`
-	WithRootDir string `toml:"-"`
+	WithAPI     string           `toml:"api"`
+	Buildpack   BpInfo           `toml:"buildpack"`
+	Order       Order            `toml:"order"`
+	WithRootDir string           `toml:"-"`
+	Targets     []TargetMetadata `toml:"targets"`
+	Stacks      []StackMetadata  `tome:"stacks"` // just for backwards compat so we can check if it's the bionic stack, which we translate to a target
+
+}
+
+type StackMetadata struct {
+	ID string `toml:"id"`
+}
+
+type TargetPartial struct {
+	Os          string `json:"os" toml:"os"`
```

Comment:
```suggestion
	OS          string `json:"os" toml:"os"`
```

Reply:
(I think this is more meticulously correct but I can't remember where I read this)

Reply:
quick grep suggests it's more consistent.  thanks!